snuba/request/schema.py

```diff
@@ -103,7 +104,8 @@ def validate(self, value, dataset: Dataset, referrer: str) -> Request:
             }
 
         query = parse_query(query_body, dataset)
-        return Request(query, self.__setting_class(**settings), extensions, referrer)
+        request_id = uuid.uuid4().hex
```

Comment:
This can potentially be non unique, though we would use this only for observability and with limited retention so I guess we are ok with it.